### PR TITLE
feat(auth): support managed oauth app

### PR DESCRIPTION
**Pull request by tomkit**
- add `use_managed_oauth` as a developer-customer-facing configuration
- add `applyUseManagedOauth` as an internal-facing configuration
- allow users in mgmt-ui to use managed oauth apps
- developer-customers can only set `use_managed_oauth` during create, not update
- update provider endpoints to allow `use_managed_oauth` on creation, not update
- intentionally did not update the integrations endpoints

Screenshot 2023-06-19 at 1 58 19 PM

## Test Plan

[Describe test plan here]

## Deployment instructions

[Add any special deployment instructions here]

### CODE REVIEW

File: packages/core/mappers/integration.ts [hunk 1]
```diff
@@ -3,20 +3,21 @@ import {
...
-  id,
-  applicationId,
-  destinationId,
-  category,
-  providerName,
-  config,
-}: IntegrationModel): Promise<Integration> => {
+export const fromIntegrationModel = async (
+  { id, applicationId, destinationId, category, providerName, config }: IntegrationModel,
+  args: IntegrationMapperArgs = { applyUseManagedOauth: true }
```
Comment (asdfryan) [inline packages/core/mappers/integration.ts:19]: rename to stripOauthConfig and reverse

File: packages/types/integration.ts [hunk 2]
```diff
@@ -16,22 +16,43 @@ export type CRMIntegration = BaseIntegration & {
...
+  oauth: OAuthConfigDecrypted;
+  sync: IntegrationSyncConfig;
+  useManagedOauth: true;
+};
+
 export type IntegrationConfigDecrypted = {
   providerAppId: string;
   oauth: OAuthConfigDecrypted;
   sync: IntegrationSyncConfig;
+  useManagedOauth?: boolean;
```
Comment (asdfryan) [inline packages/types/integration.ts:48]: useManagedOauth?: false

File: packages/types/integration.ts [hunk 3]
```diff
@@ -16,22 +16,43 @@ export type CRMIntegration = BaseIntegration & {
...
 };
 
 export type IntegrationConfigEncrypted = {
   providerAppId: string;
   oauth: OAuthConfigEncrypted;
   sync: IntegrationSyncConfig;
+  useManagedOauth?: boolean;
```
Comment (asdfryan) [inline packages/types/integration.ts:55]: useManagedOauth?: false

File: apps/api/routes/internal/integration/index.ts [hunk 4]
```diff
@@ -31,7 +31,9 @@ export default function init(app: Router): void {
       req: Request<GetIntegrationsPathParams, GetIntegrationsResponse, GetIntegrationsRequest>,
       res: Response<GetIntegrationsResponse>
     ) => {
-      const integrations = await integrationService.list(req.supaglueApplication.id);
+      const integrations = await integrationService.list(req.supaglueApplication.id, {
```
Comment (asdfryan) [inline apps/api/routes/internal/integration/index.ts:34]: we should also update the mgmt API

File: packages/core/mappers/integration.ts (hunk 1 above)
Comment (tomkit) [reply packages/core/mappers/integration.ts]: it empties the field value, but not the key, named it `hideManagedOauthConfig`

File: packages/types/integration.ts (hunk 2 above)
Comment (tomkit) [reply packages/types/integration.ts:29]: the oauth config signature doesn't change based on this, let's keep it loose so we don't need to do a type tree shake

**Review by asdfryan — APPROVED** [review]
This seems fine to me, but you'll have to re-implement the FE after my PR I think.

**Pull request merged**